### PR TITLE
General updates to the library

### PR DESCRIPTION
**Pull request by giocalitri**
part of #8

### CODE REVIEW

File: gitreload/config.py [hunk 1]
```diff
@@ -1,38 +1,45 @@
...
-    os.environ.get('GITRELOAD_CONFIG', ''),
-    os.path.join(os.getcwd(), 'gr.env.json'),
-    os.path.join(os.path.expanduser('~'), '.gr.env.json'),
-    '/etc/gr.env.json',
-]
+MINUTE = 60  # seconds
 
 
-def configure_logging(level_override=None):
+class Config(object):
```
Comment (gsidebo) [inline gitreload/config.py]: General question: have you removed the ability to configure the app via a JSON file, and now it's only possible to configure via OS environment variables? I personally find JSON config a _lot_ easier to manage. Seems like we could have both if we attempted to load a JSON file from some path(s) before defining this class, then adding a helper method that tries to get the value from the OS environment, then the JSON config file dict if that doesn't exist, then a default value if that doesn't exist.

File: gitreload/config.py [hunk 2]
```diff
@@ -1,38 +1,45 @@
...
+    LINKED_REPOS = os.environ.get('LINKED_REPOS', {})
+    ALSO_CLONE_REPOS = os.environ.get('ALSO_CLONE_REPOS', {})
+    NUM_THREADS = int(os.environ.get('NUM_THREADS', 1))
+    LOG_LEVEL = os.environ.get('LOG_LEVEL', None)
+    HOSTNAME = platform.node().split('.')[0]
+    LOG_FORMATTER = ('%(asctime)s %(levelname)s %(process)d [%(name)s] '
+                     '%(filename)s:%(lineno)d - '
+                     '{hostname}- %(message)s').format(hostname=HOSTNAME)
+    LOG_FILE_PATH = os.environ.get('LOG_FILE_PATH', '')
+    SUBPROCESS_TIMEOUT = int(os.environ.get('SUBPROCESS_TIMEOUT', 60)) * MINUTE
```
Comment (gsidebo) [inline gitreload/config.py]: Would probably help to call this variable `SUBPROCESS_TIMEOUT_MINUTES` or something. I typically assume that a timeout value is in seconds

File: gitreload/config.py (hunk 1 above)
Comment (giocalitri) [reply gitreload/config.py]: I talked about it with Tobias and he agreed on the change

File: gitreload/config.py (hunk 2 above)
Comment (giocalitri) [reply gitreload/config.py]: good point: I'll change that

Comment (giocalitri) [reply gitreload/config.py]: I'll change only the environment variable name and keep the config one as is, given that I translate that in seconds.

File: gitreload/config.py (hunk 1 above)
Comment (gsidebo) [reply gitreload/config.py]: OK

**Comment by giocalitri** [comment]
@gsidebo Done

**Comment by pdpinch** [comment]
The PR has been merged. How/can we release this @mitodl/devops ?

**Comment by shaidar** [comment]
This will be part of the AMI build that I'm currently working on and should have in QA today.

**Pull request merged**